utils/paginated.js

```diff
@@ -0,0 +1,14 @@
+async function* paginated(perform, limit = Infinity) {
+  let nextPage = undefined;
+  for (let i = 0; nextPage != null && i < limit; i++) {
```

Comment:
It still doesn't work. The condition nextPage != null is `false` for nextPage === `undefined`.

Reply:
I would use do...while loop.
```
let i = 0;
do {
  const result = await perform(nextPage);
    if (result.isErr()) {
      throw result.error;
    } else {
      nextPage = result.value.nextPage;
      yield result.value;
    }
    i++;
} while (nextPage != null && i < limit);
```

Reply:
:+1: